astropy/utils/tests/test_state.py

```diff
@@ -0,0 +1,31 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import pytest
+
+from astropy.utils.state import ScienceState
+
+
+def test_ScienceState_and_Context():
+    """
+    Tests a ScienceState and spawned contexts.
+    """
+    class MyState(ScienceState):
+        _value = "A"
+        _state = dict(foo="bar")
+
+    state = {"foo": "bar"}
+
+    # test created ScienceState
+    assert MyState.get() == "A"
+    assert MyState.validate("B") == "B"
+    assert MyState._state == state
+
+    # test setting
```

Comment:
Can we write this as `with MyState.set("B")` - and then test that `.get() == 'B'` but that it returns to 'A' after the context? Our tests should check the public API, not the implementation!

Reply:
Agreed. Done.